src/classes/GS_ApplicationStatusController.cls

```diff
@@ -0,0 +1,176 @@
+     */
+    private Date getApplicationDate(Map<String, Object> testingConfiguration) {
+        
+        if (isScratchOrg() && testingConfiguration != null) { 
+            return testingConfiguration.get(APPLICATION_PARAMETER) == null ? null: Date.parse(testingConfiguration.get(APPLICATION_PARAMETER).toString());
+        }
+
+        try {
+            return System.FeatureManagement.checkPackageDateValue(APPLICATION_DATE);
+        } catch (System.NoDataFoundException ex) {
+            return Date.Today();
+        }
+    }
+
+    /**
+     * @description Retrieves the Trial expiration date.
+     * On prod this is populated quering the Organization object.
+     * For QA purpose this values can be overriden if this is running on a Scratch org and there is a value in the custom setting Testing_Configuration__c
+     * @param  overrideValuesForTesting Boolean true if usage of Testing_Configuration__c custom setting is desired
+     * @param  testingConfiguration     Map<String, Object> parsed TestingConfiguration object
+     * @param  org                      Organization the Organization object, the query should have included the TrialExpiration field 
+     * @return                          DateTime with the expirationDate.
+     */
+    private DateTime getExpirationDate(Boolean overrideValuesForTesting,Map<String,Object> testingConfiguration, Organization org) {
+        
+        if (overrideValuesForTesting) {
+            return  (testingConfiguration.get(TRIAL_EXPIRATION_PARAMETER) == null ? null : 
+                    DateTime.parse(testingConfiguration.get(TRIAL_EXPIRATION_PARAMETER).toString()));
+        } else {
+            return org.TrialExpirationDate;
+        }
+        
+    }
+
+    /**
+     * @description Returns the parsed Testing_Configuration.
+     * @return   Map<String, Object> Parsed Testing_Configuration Custom_Setting. it is a json format.
+     * Examples: 
+     * {"Sandbox":false} simulates an Active org.
+     * {"Sandbox":false,"TrialExpiration":"10/15/2020, 00:01 am”} simulates a trial org which is not applied for licenses.
+     * {"Sandbox":false,"TrialExpiration":"10/15/2020","Application":"10/02/2020"} Simulates a trial org which has applied for licenses.
+     */
+    private Map<String, Object> getTestingConfiguration() {
+        Testing_Configuration__c[] tc = [SELECT Configuration__c 
+                                         FROM   Testing_Configuration__c];
+        if (!tc.IsEmpty()) {
+            Map<String, Object> parsedConfiguration = (Map<String, Object>) JSON.deserializeUntyped(tc[0].Configuration__c);
+            return parsedConfiguration;
+        }
+        return null;
+    }
+
+    /**
+     * Returns if it's running on a Scratch org.
+     * @return true if code runs on a Scratch org, false otherwise
+     */
+    private Boolean isScratchOrg() {
+        Organization org = getOrgInformation();
```

Comment:
There is probably a better implementation of this method in `UTIL_Namespace.isScratchOrg()`. We worked with core to verify an acceptable way to validate this.